zebra-network/src/protocol/external/message.rs

```diff
@@ -340,20 +340,29 @@ pub enum Message {
     FilterClear,
 }
 
+/// The maximum size of the rejection message
+const MAX_REJECT_MESSAGE_LENGTH: usize = 12;
+/// The maximum size of the rejection reason
+const MAX_REJECT_REASON_LENGTH: usize = 100;
+
 impl<E> From<E> for Message
 where
     E: Error,
 {
     fn from(e: E) -> Self {
+        let mut message = e.to_string();
+        message.truncate(MAX_REJECT_MESSAGE_LENGTH);
         Message::Reject {
-            message: e.to_string(),
+            message,
 
             // The generic case, impls for specific error types should
             // use specific varieties of `RejectReason`.
             ccode: RejectReason::Other,
 
             reason: if let Some(reason) = e.source() {
-                reason.to_string()
+                let mut reason = reason.to_string();
+                reason.truncate(MAX_REJECT_REASON_LENGTH);
```

Comment:
Can we escape any weird characters before truncating the string?
https://doc.rust-lang.org/std/primitive.str.html#method.escape_default

Error message strings might contain weird characters, because they are generated from untrusted peer data. So we should do something to sanitize them.

If we don't escape weird characters, our limits might not work like we expect. Because `truncate()` limits the length in UTF-8 characters, but `zcashd` limits the length in bytes. (Some UTF-8 characters can take up multiple bytes.)

Here is how we convert the string to bytes when we serialize the message:
https://github.com/ZcashFoundation/zebra/blob/e7094201604c2f473145169a10c23a703a57c96d/zebra-chain/src/serialization/zcash_serialize.rs#L163

Here are some examples of multi-byte UTF-8 characters:
https://doc.rust-lang.org/std/str/fn.from_utf8.html#examples